zerver/tests/test_auth_backends.py

```diff
@@ -1646,8 +1651,69 @@ def register_extra_endpoints(self, requests_mock: responses.RequestsMock,
 
         self.email_data = email_data
 
-    def get_account_data_dict(self, email: str, name: str) -> Dict[str, Any]:
-        return dict(email=email, name=name)
+    def get_account_data_dict(self, email: str, name: str, user_avatar_url: str=None) -> Dict[str, Any]:
+        return dict(email=email, name=name, user_avatar_url=user_avatar_url)
+
+    def get_log_into_subdomain(self, data: Dict[str, Any], *, subdomain: str='zulip',
+                               force_token: Optional[str]=None) -> HttpResponse:
+        if force_token is None:
+            token = store_login_data(data)
+        else:
+            token = force_token
+        url_path = reverse('zerver.views.auth.log_into_subdomain', args=[token])
+        return self.client_get(url_path, subdomain=subdomain)
+
+    #do tests when the user has an invite to join
+    @override_settings(TERMS_OF_SERVICE=None)
+    def test_use_github_avatar(self) -> None:
+        """If the user doesn't exist yet, social auth can be used to register an account"""
+        email = 'newuser@zulip.com'
+        name = 'Full Name'
+        subdomain = 'zulip'
+        mock_user_avatar_url = 'https://github.com/user/mock_avatar_url/image.jpeg'
+        realm = get_realm("zulip")
+        account_data_dict = self.get_account_data_dict(email=email, name=name, user_avatar_url=mock_user_avatar_url)
+        result = self.social_auth_test(account_data_dict,
+                                       expect_choose_email_screen=True,
+                                       subdomain=subdomain, is_signup='1')
+        self.assertEqual(result.status_code, 302)
+        self.assertIn('/login/subdomain', result.url)
+
+        account_data_dict['is_signup'] = True
+        account_data_dict['next'] = ''
+        account_data_dict['subdomain'] = subdomain
+
+        # Confirm go to registration
```

Comment:
For the test, I think you should use `stage_two_of_registration` and add to it the ability to pass in an avatar.